src/main/java/BasketballGame/GameRegister.java

```diff
@@ -0,0 +1,221 @@
+    // it's a lot easier to do it with ArrayList.
+    private final ArrayList<Game> game;
+    private LocalDateTime localDateTime = LocalDateTime.now();
+    private final DateTimeFormatter myFormatObj = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    public GameRegister() {
+        this.game = new ArrayList<Game>();
+    }
+
+    public void addGame(Game game) {
+        this.game.add(game);
+    }
+
+    public void printGameRegister() {
+        for (int i = 0; i < game.size(); i++) {
+            printDate(i);
+        }
+    }
+
+    public void printPastGames() {
+        for (int i = 0; i < game.size(); i++) {
+            if (game.get(i).getGameTime().isBefore(localDateTime)) {
+                printDateWithScores(i);
+            }
+        }
+    }
+
+    public void printFutureGames() {
+        for (int i = 0; i < game.size(); i++) {
+            if (game.get(i).getGameTime().isAfter(localDateTime)) {
+                printDate(i);
+            }
+        }
+    }
+
+    public void getGamesOfThisWeek() {
+        LocalDateTime monday = localDateTime.with(previousOrSame(MONDAY));
+        LocalDateTime sunday = localDateTime.with(nextOrSame(SUNDAY));
+        for (int i = 0; i < game.size(); i++) {
+            if (monday.getDayOfMonth() <= game.get(i).getGameTime().getDayOfMonth() && sunday.getDayOfMonth() >= game.get(i).getGameTime().getDayOfMonth()) {
+                printDate(i);
+            }
+        }
+    }
+
+    public void getGamesAtDateTime(DateTime dateTime) {
+        int year = dateTime.getYear(),month = dateTime.getMonth(), day = dateTime.getDay(), hour = dateTime.getHour();
+        for (int i = 0; i < game.size(); i++) {
+            int gameYear = game.get(i).getGameTime().getYear(), gameMonth = game.get(i).getGameTime().getMonth().getValue();
+            int gameDay = game.get(i).getGameTime().getDayOfMonth(), gameHour = game.get(i).getGameTime().getHour();
+            if (year != 0 && month != 0 && day != 0 && hour != 0) {
+                if (year == gameYear && month == gameMonth && day == gameDay && hour == gameHour) {
+                    printDate(i);
+                }
+            } else if (year != 0 && month != 0 && day != 0) {
+                if (year == gameYear && month == gameMonth && day == gameDay) {
+                    printDate(i);
+                }
+            } else if (year != 0 && month != 0) {
+                if (year == gameYear && month == gameMonth) {
+                    printDate(i);
+                }
+            }
+        }
+    }
+
+    public void printTodaysGames() {
+        for (int i = 0; i < game.size(); i++) {
+            if (localDateTime.getDayOfMonth() == (game.get(i).getGameTime().getDayOfMonth())) {
+                printDate(i);
+            }
+        }
+    }
+
+    public void printGameInformation (Game game) {
+        topOrBottom(game,"Top");
+        printRow(game);
+        topOrBottom(game,"Bottom");
+    }
```

Comment:
Don't know if i understood correctly, but I guess this kinda looks what you asked for printing game information (Hard part of the Challenge).